### PR TITLE
Add GET /_api/bookmarks.list API

### PR DESCRIPTION
**Pull request by hirakiuc**
Add `/_api/bookmarks.list` API to fetch bookmarks.

```json
$ curl -X GET "http://localhost:3000/_api/bookmarks.list?access_token=MY_SECRET_TOKEN&limit=5&offset=0" | jq '.'
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  3100  100  3100    0     0  85406      0 --:--:-- --:--:-- --:--:-- 86111
{
  "meta": {
    "total": 3,
    "limit": 5,
    "offset": 0
  },
  "data": [
    {
      "_id": "5b152e55d8fe3503492e90df",
      "user": "5b123ea3403dd50264d0fec4",
      "page": {
        "_id": "5b133b6b83558c0266a8454a",
        "redirectTo": null,
        "updatedAt": "2018-06-03T00:50:51.144Z",
        "lastUpdateUser": "5b123ea3403dd50264d0fec4",
        "creator": "5b123ea3403dd50264d0fec4",
        "path": "/user/hirakiuc/memo/2018/06/03",
        "__v": 1,
        "revision": {
          "_id": "5b133b6b83558c0266a8454b",
          "author": {
            "_id": "5b123ea3403dd50264d0fec4",
            "email": "hirakiuc@gmail.com",
            "username": "hirakiuc",
            "name": "Hirakiuchi Daisuke",
            "admin": true,
            "createdAt": "2018-06-02T06:52:19.189Z",
            "status": 2,
            "lang": "en"
          },
          "body": "# memo/2018/06/03\r\n\r\nsplatoon2のガチマッチ、ちょっと進み始めた。\r\n\r\nまだB だから大丈夫ぽい。\r\n\r\nギア集めしないと",
          "path": "/user/hirakiuc/memo/2018/06/03",
          "__v": 0,
          "createdAt": "2018-06-03T00:50:51.135Z",
          "format": "markdown"
        },
        "createdAt": "2018-06-03T00:50:51.066Z",
        "commentCount": 0,
        "seenUsers": [
          "5b123ea3403dd50264d0fec4"
        ],
        "liker": [],
        "grantedUsers": [
          "5b123ea3403dd50264d0fec4"
        ],
        "grant": 1,
        "status": "published",
        "id": "5b133b6b83558c0266a8454a"
      },
      "__v": 0,
      "createdAt": "2018-06-04T12:19:33.677Z"
    },
    {
      "_id": "5b152e50d8fe3503492e90de",
      "user": "5b123ea3403dd50264d0fec4",
      "page": {
(...snipped...)
        },
        "createdAt": "2018-06-02T06:54:11.200Z",
        "commentCount": 0,
        "seenUsers": [
          "5b123ea3403dd50264d0fec4"
        ],
        "liker": [],
        "grantedUsers": [
          "5b123ea3403dd50264d0fec4"
        ],
        "grant": 1,
        "status": "published",
        "id": "5b123f13403dd50264d0fec7"
      },
      "__v": 0,
      "createdAt": "2018-06-04T12:19:28.426Z"
    },
    {
      "_id": "5b138f338d2a47026b50cbfb",
      "user": "5b123ea3403dd50264d0fec4",
      "page": {
(...snipped...)
        },
        "createdAt": "2018-06-02T06:54:37.877Z",
        "commentCount": 0,
        "seenUsers": [
          "5b123ea3403dd50264d0fec4"
        ],
        "liker": [],
        "grantedUsers": [
          "5b123ea3403dd50264d0fec4"
        ],
        "grant": 1,
        "status": "published",
        "id": "5b123f2d403dd50264d0fec9"
      },
      "__v": 0,
      "createdAt": "2018-06-03T06:48:19.287Z"
    }
  ],
  "ok": true
}
```

## Motivation

same with https://github.com/crowi/crowi/pull/293.

### Question

The meta field of `/_api/bookmarks.list` API response is not same with `/_api/pages.list` API.

I think that API response should have same meta fields.

What do you think about meta field of API response?

### CODE REVIEW

**Comment by hirakiuc** [comment]
This PR should have same task about https://github.com/crowi/crowi/pull/293#issuecomment-394422516.

**Comment by hirakiuc** [comment]
Rebased and Updated codes. 🙇

**Comment by hirakiuc** [comment]
Oh, I'll check this in this weekend!

**Comment by hirakiuc** [comment]
Rebased and fixed lint errors. 👍

File: lib/models/bookmark.js [hunk 1]
```diff
@@ -59,6 +59,57 @@ module.exports = function(crowi) {
     })
   }
 
+  bookmarkSchema.statics.findByUserId = function(userId, option) {
+    var User = crowi.model('User')
```
Comment (lightnet328) [inline lib/models/bookmark.js]: `User` is not used.
Please delete this line.

File: lib/models/bookmark.js [hunk 2]
```diff
@@ -59,6 +59,57 @@ module.exports = function(crowi) {
...
+          return reject(err)
+        }
+
+        return resolve(count)
+      })
+    })
+
+    return Promise.all([finder, counter])
+      .then(function(results) {
+        bookmarks = results[0]
```
Comment (lightnet328) [inline lib/models/bookmark.js]: Please use `var` to define variables.
In this case, You can write as follows.
```javascript
.then(function([bookmarks, count]) {
```

**Pull request merged**